src/puppetlabs/trapperkeeper/services/scheduler/scheduler_service.clj

```diff
@@ -19,6 +19,33 @@
       (tk/service-context)
       :jobs))
 
+(defn- queue-job!
+  ([this id]
+   (queue-job! this id {}))
+  ([this id opts]
+   (let [result (merge {:id id} opts)]
+    (swap! (get-jobs this) conj result)
+    result)))
+
+(defn- dequeue-job!
+  [this id]
+  (if-let [item (first (filter #(= id (:id %)) @(get-jobs this)))]
+    (swap! (get-jobs this) disj item)))
+
+(defn- wrap-remove-job-by-after-id!
```

Comment:
My intent was to prevent memory leaks with the "after" jobs identifiers.  They will remain in the job set after the job completes, if they are not explicitly removed (which is what I'm doing in the finally block).   There is a chicken-and-egg issue with using the existing job identifier -- it doesn't get created until after the job is scheduled.    I'll add some docs strings and play with the naming, maybe that will make it clearer.